src/Npgsql/Internal/TypeHandlers/RangeHandler.cs

```diff
@@ -178,11 +178,11 @@ public override int ValidateObjectAndGetLength(object? value, ref NpgsqlLengthCa
     public override Task WriteObjectWithLength(object? value, NpgsqlWriteBuffer buf, NpgsqlLengthCache? lengthCache, NpgsqlParameter? parameter, bool async, CancellationToken cancellationToken = default)
         => value switch
         {
-            NpgsqlRange<TSubtype1> converted => WriteWithLength(converted, buf, lengthCache, parameter, async, cancellationToken),
-            NpgsqlRange<TSubtype2> converted => WriteWithLength(converted, buf, lengthCache, parameter, async, cancellationToken),
+            NpgsqlRange<TSubtype1> converted => ((INpgsqlTypeHandler<NpgsqlRange<TSubtype1>>)this).WriteWithLength(converted, buf, lengthCache, parameter, async, cancellationToken),
+            NpgsqlRange<TSubtype2> converted => ((INpgsqlTypeHandler<NpgsqlRange<TSubtype2>>)this).WriteWithLength(converted, buf, lengthCache, parameter, async, cancellationToken),
```

Comment:
Any particular reason for these casts?

Reply:
They reduce WriteWithLength<TAny> code bloat by redirecting it to the new extension method on the INpgsqlTypeHandler that mirrors the behavior (I'll push a commit to call this from within WriteWithLength<TAny> to keep its behavior in sync)

Reply:
Nvm that would duplicate checks and add another state machine to the common path.